crates/bevy_ecs/src/schedule/config.rs

```diff
@@ -496,12 +555,18 @@ impl IntoSystemConfigs<()> for SystemConfigs {
         self
     }
 
-    fn distributive_run_if<M>(mut self, condition: impl Condition<M> + Clone) -> SystemConfigs {
+    fn distributive_run_if<M>(mut self, condition: impl Condition<M> + Clone) -> NodeConfigs<T>
+    where
+        T: NodeType,
+    {
```

Comment:
```suggestion
    fn distributive_run_if<M>(mut self, condition: impl Condition<M> + Clone) -> NodeConfigs<T> {
```
This is a redundant bound since the trait already specifies `T: NodeType`.